lib/service/connect.go

```diff
@@ -529,6 +529,12 @@ func (process *TeleportProcess) firstTimeConnect(role types.SystemRole) (*Connec
 		if err != nil {
 			return nil, trace.Wrap(err)
 		}
+
+		defer func() {
+			if !reused && err != nil {
+				clt.Close()
+			}
+		}()
```

Comment:
There's only two error return paths that need a `connector.Close()` tho, is that worse than a conditional defer?